src/impl/windows/external_commit_helper.cpp

```diff
@@ -25,19 +25,40 @@ using namespace realm;
 using namespace realm::_impl;
 using namespace realm::util;
 
-static std::wstring create_condvar_sharedmemory_name(std::string realm_path) {
+static std::string normalize_realm_path_for_windows_kernel_object_name(std::string realm_path) {
+    // windows named objects names should not contain backslash
     std::replace(realm_path.begin(), realm_path.end(), '\\', '/');
-    return L"Local\\Realm_ObjectStore_ExternalCommitHelper_SharedCondVar_" + std::wstring(realm_path.begin(), realm_path.end());
+
+    // always use lowercase for the drive letter as a win32 named objects name
+    auto position = realm_path.find(':');
+    if (position != std::string::npos && position > 0) {
+        realm_path[position - 1] = tolower(realm_path[position - 1]);
+    }
+
+    return realm_path;
+}
+
+static std::string create_condvar_sharedmemory_name(std::string realm_path) {
+    realm_path = normalize_realm_path_for_windows_kernel_object_name(realm_path);
+
+    std::string name("Local\\Realm_ObjectStore_ExternalCommitHelper_SharedCondVar_");
+    name.append(realm_path);
+    return name;
 }
 
 ExternalCommitHelper::ExternalCommitHelper(RealmCoordinator& parent)
 : m_parent(parent)
-, m_condvar_shared(create_condvar_sharedmemory_name(parent.get_path()).c_str())
+, m_condvar_shared(create_condvar_sharedmemory_name(parent.get_path()))
 {
-    m_mutex.set_shared_part(InterprocessMutex::SharedPart(), parent.get_path(), "ExternalCommitHelper_ControlMutex");
-    m_commit_available.set_shared_part(m_condvar_shared.get(), parent.get_path(),
-                                       "ExternalCommitHelper_CommitCondVar",
-                                       std::filesystem::temp_directory_path().u8string());
+    m_mutex.set_shared_part(InterprocessMutex::SharedPart(), 
+        normalize_realm_path_for_windows_kernel_object_name(parent.get_path()), 
```

Comment:
Ideally this "path to name" normalization should happen in InterProcessMutex and InterProcessCondvar since they take a path as argument
//cc: @finnschiermer